stock_picking_mass_action/models/stock_picking.py

```diff
@@ -18,35 +18,32 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
+
 import logging
 
 from openerp.osv import orm
+from openerp.models import Model, api
 
 _logger = logging.getLogger(__name__)
 
 
-class stock_picking(orm.Model):
-
+class StockPicking(Model):
     _inherit = 'stock.picking'
 
-    def check_assign_all(self, cr, uid, ids=None, context=None):
+    @api.model
+    def check_assign_all(self):
         """ Try to assign confirmed pickings """
-        if isinstance(ids, (int, long)):
-            ids = [ids]
-        if ids is None:
-            domain = [('type', '=', 'out'),
-                      ('state', '=', 'confirmed')]
-            ids = self.search(cr, uid, domain,
-                              order='min_date',
-                              context=context)
+        spt_obj = self.env['stock.picking.type']
+        spt_out_lst = spt_obj.search([('code', '=', 'outgoing')])
+        spt_out_ids = [x.id for x in spt_out_lst]
+        domain = [('picking_type_id', 'in', spt_out_ids),
+                  ('state', '=', 'confirmed')]
+        records = self.search(domain, order='min_date')
 
-        for picking_id in ids:
+        for record in records:
             try:
-                self.action_assign(cr, uid, [picking_id], context)
+                record.action_assign()
             except orm.except_orm:
```

Comment:
Now you should check also `exceptions.Warning` or any of the possible exceptions. Maybe a general except?